code/src/hooks/use-infinite-search/useInfiniteSearch.ts

```diff
@@ -0,0 +1,75 @@
+import { skipToken } from "@reduxjs/toolkit/query/react";
+import { useEffect, useRef, useState } from "react";
+
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useDebouncedValue from "@/hooks/use-debounced-value/useDebouncedValue";
+import { useGetUserProductsBySearchQuery } from "@/store/api/productsApi";
+import { GetUserProductsBySearchQueryResponse } from "@/types/product.types";
+
+const RESULTS_PER_PAGE = 10;
+const MIN_QUERY_LENGTH = 4;
+const DEBOUNCE_DELAY_MS = 500;
+
+const useInfiniteSearch = (query: string) => {
+  const [currentPage, setCurrentPage] = useState(0);
+  const [searchProducts, setSearchProducts] =
+    useState<GetUserProductsBySearchQueryResponse>();
```

Comment:
To make it more reusable we can also use generic and rename state:

```suggestion
    const [pageableValue, setPageableValue] = useState<PageableResponse<T>>();
```

Reply:
Discussed in voice